llama_github/rag_processing/rag_processor.py

```diff
@@ -290,173 +291,182 @@ def _split_content_into_chunks(self, content: str, language: Optional[str] = Non
         chunks = splitter.split_text(content)
         return chunks
 
-    def _arrange_issue_search_result(self, issue_search_result: dict) -> dict:
+    def _arrange_issue_search_result(self, issue_search_result: dict) -> List[Dict[str, str]]:
         """
-        arrange the result of issue search
+        Arrange the result of Issue search with metadata.
 
         Args:
-            issue_search_result (dict): the result of issue search.
+            _arrange_issue_search_result (dict): The result of Issue search.
 
         Returns:
-            dict: the arranged result of issue search.
+            List[Dict[str, str]]: The arranged result of Issue search with metadata.
         """
         arranged_results = []
 
         for result in issue_search_result:
             content = result['issue_content']
+            url = result.get('url', '')
 
             # Split content into chunks
             chunks = self._split_content_into_chunks(
                 content, max_tokens=config.get('issue_chunk_size'))
 
             for chunk in chunks:
-                arranged_results.append(chunk)
+                arranged_results.append({'context': chunk, 'url': url})
 
         return arranged_results
 
-    def _arrange_repo_search_result(self, repo_search_result: dict) -> dict:
+    def _arrange_repo_search_result(self, repo_search_result: dict) -> List[Dict[str, str]]:
         """
-        arrange the result of repo search
+        Arrange the result of Repo search with metadata.
 
         Args:
-            repo_search_result (dict): the result of repo search.
+            _arrange_repo_search_result (dict): The result of Repo search.
 
         Returns:
-            dict: the arranged result of repo search.
+            List[Dict[str, str]]: The arranged result of Repo search with metadata.
         """
         arranged_results = []
 
         for result in repo_search_result:
             content = result['content']
+            url = result.get('url', '')
```

Comment:
For the repo search result, please notice that the initial search result (refer to `github_rag.py` lines 330-334) only contains the following parameters:

```python
results_with_index.append({
    'index': index,
    'full_name': repository.full_name,
    'content': repo_readme,
})
```

We need to compose the final URL by using `full_name`. Please kindly help to rewrite the code accordingly.